File: tests/fixtures/repomap/sample.m
#import <Foundation/Foundation.h>
#import "AppDelegate.h"

@interface UserService : NSObject
- (void)getUser:(int)userId;
- (void)listUsers;
@end

@implementation UserService
- (void)getUser:(int)userId {
    NSLog(@"Getting user %d", userId);
}
- (void)listUsers {
    NSLog(@"Listing users");
}
@end

@protocol Repository
- (void)save:(id)entity;
- (id)findById:(int)entityId;
@end

void createApp(NSDictionary *config) {
    NSLog(@"Creating app");
}

File: tests/fixtures/repomap/sample.swift
import Foundation
import UIKit

public class UserService {
    public func getUser(id: Int) -> User? {
        return nil
    }

    private func validate(id: Int) -> Bool {
        return id > 0
    }
}

public protocol Repository {
    func save(entity: Any)
    func find(id: Int) -> Any?
}

public struct UserData {
    var id: Int
    var name: String
}

func internalHelper() -> Void {
    // not public
}

public func createApp(config: [String: Any]) -> UserService {
    return UserService()
}
